w04/test/gad/simplesort/MedianPivotDistributedTest.java

```diff
@@ -69,4 +69,75 @@ private static Stream<Arguments> fromEquals0() {
                         0,
                         30));
     }
+
+    private static Stream<Arguments> edgeCases() {
+        return Stream.of(
+                Arguments.of(
+                        new int[]{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+                        4,
+                        3,
+                        0,
+                        9),
+                Arguments.of(
+                        new int[]{9, 1, 8, 5, 2, 3, 5, 1, 0, 7},
+                        6,
+                        5,
+                        0,
+                        9),
+                Arguments.of(
+                        new int[]{9, 1, 8, 5, 2, 3, 5, 1, 0, 7},
+                        5,
+                        5,
+                        1,
+                        9),
+                Arguments.of(
+                        new int[]{9, 6, 2, 2, 2},
+                        4,
+                        5,
+                        4,
+                        4),
+                Arguments.of(
+                        new int[]{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+                        4,
+                        3,
+                        0,
+                        9),
+                Arguments.of(
+                        new int[]{9, 1, 8, 5, 2, 3, 5, 1, 0, 7},
+                        6,
+                        5,
+                        0,
+                        9),
+                Arguments.of(
+                        new int[]{9, 1, 8, 5, 2, 3, 5, 1, 0, 7},
+                        5,
+                        5,
+                        1,
+                        9),
+                Arguments.of(
+                        new int[]{9, 1, 8},
+                        2,
+                        3,
+                        0,
+                        2),
+                Arguments.of(
+                        new int[]{9, 1},
+                        0,
+                        2,
+                        0,
+                        1),
+                Arguments.of(
+                        new int[]{9},
+                        0,
+                        1,
+                        0,
+                        0),
```

Comment:
Thanks for your tests! I'm failing these two cases but have 100% of the points on this exercise.

Reply:
Does it work for you when you change the `numbersOfConsideredElements` to 3 or 5?
I sadly cannot reproduce it as my implementation also works with these values.

Reply:
Yes, with `3` and `5` is works for me 👍

Reply:
I updated the branch